pkg/webhook/cspc.go

```diff
@@ -527,11 +528,18 @@ func (p *PoolOperations) ValidateScaledown() (bool, string) {
 				break
 			}
 		}
+
+		ls := &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				types.CStorPoolClusterLabelKey: p.OldCSPC.Name,
+				types.HostNameLabelKey:         oldPool.NodeSelector[types.HostNameLabelKey],
```

Comment:
Since `NodeSelector` is populated by `[Users]`(https://github.com/openebs/cstor-operators/blob/74b6a3a04e58b45136bbb077ee1dcba0f3c47e1b/pkg/cspc/algorithm/build_csp.go#L71) it will not guarantee that we can fetch `kubernetes.io/hostName` value using NodeSelector(Thinking about a case where users provide some other unique key to identify the node).

Reply:
The `kubernetes.io/hostname` label -- whether provided by the user or not -- finally CSPC-operator finds out the node from the labels provided in the CSPC and adds a `kubernetes.io/hostname` label on CSPI by itself.

Reply:
But, will relook on that workflow and test it. If there is any fix required -- that will go in cspc-operator.

Reply:
Got your point... This is a nice catch. Done

Reply:
Good catch !!!